Source/Scene/Primitive.js

```diff
@@ -1784,6 +1882,21 @@ define([
         updateAndQueueCommandsFunc(this, frameState, this._colorCommands, this._pickCommands, this.modelMatrix, this.cull, this.debugShowBoundingVolume, twoPasses);
     };
 
+    var offsetBoundingSphereScratch1 = new BoundingSphere();
+    var offsetBoundingSphereScratch2 = new BoundingSphere();
+    function transformBoundingSphere(boundingSphere, offset, extend) {
+        if (extend) {
+            var origBS = BoundingSphere.clone(boundingSphere, offsetBoundingSphereScratch1);
+            var offsetBS = BoundingSphere.clone(boundingSphere, offsetBoundingSphereScratch2);
+            offsetBS.center = Cartesian3.add(offsetBS.center, offset, offsetBS.center);
+            boundingSphere = BoundingSphere.union(origBS, offsetBS, boundingSphere);
```

Comment:
This should take into account what the offset type is. If it's ALL, then the whole bounding sphere moves. If it's TOP, then it would be the union like you have here.

Reply:
That's what the `extend` param to the function does